service/src/main/resources/db/migration/V251__assistance_need_decision_read.sql

```diff
@@ -0,0 +1,2 @@
+ALTER TABLE assistance_need_decision
+  ADD COLUMN read_guardian_ids uuid[] DEFAULT array[]::uuid[];
```

Comment:
I think this data model might have some issues when guardian information changes...if somebody becomes a guardian for a child, won't they get "unread decision" notifications of all the child's decisions since the beginning of time, since the new guardian is not in this list in old decisions?

My quick 1 minute hunch is that this would work better using the opposite semantics: have a list of *unread* guardian ids, and write that during some decision status change event (decision created, or when status changes to accepted/rejected?). Reading the notification should then remove the guardian id from the list, and new guardians won't suddenly get notifications for old stuff because the list is essentially a "snapshot" of guardians at one point of time.

Reply:
I had the opposite thinking that when guardians are changed, the new guardians would see the old decisions as unread, but I do agree that in practice it's probably not useful to see very old decisions as new notifications. I'll switch the logic around as you mentioned 👍